src/deepsparse/transformers/pipelines.py

```diff
@@ -0,0 +1,836 @@
+    """
+
+    @abstractmethod
+    def transform(self, X):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def predict(self, X):
+        raise NotImplementedError()
+
+
+class Pipeline(_ScikitCompat):
+    """
+    The Pipeline class is the class from which all pipelines inherit.
+    Refer to this class for methods shared across different pipelines.
+    This base Pipeline class provides support for multiple inference engine backends.
+
+    Base class implementing pipelined operations.
+    Pipeline workflow is defined as a sequence of the following operations:
+
+        Input -> Tokenization -> Model Inference ->
+        Post-Processing (task dependent) -> Output
+
+    Pipeline supports running with the DeepSparse engine or onnxruntime.
+
+    :param model: loaded inference engine to run the model with, can be a
+        deepsparse Engine or onnxruntime InferenceSession
+    :param tokenizer: tokenizer to be used for preprocessing
+    :param config: transformers model config for this model
+    :param engine_type: name of inference engine that is used. Options are
+        deepsparse and onnxruntime
+    :param max_length: maximum sequence length to set for model inputs by default.
+        default value is 128
+    :param input_names: list of input names to the neural network
+    :param args_parser: Reference to the object in charge of parsing supplied
+        pipeline parameters. A default is provided if None
+    :param binary_output: if True, stores outputs as pickled binaries to avoid
+        storing large amount of textual data. Default is False
+    """
+
+    default_input_names = None
+
+    def __init__(
+        self,
+        model: Union[Engine, InferenceSession],
+        tokenizer: PreTrainedTokenizer,
+        config: PretrainedConfig,
+        engine_type: str,
+        max_length: int = 128,
+        input_names: Optional[List[str]] = None,
+        args_parser: ArgumentHandler = None,
+        binary_output: bool = False,
+    ):
+
+        _validate_transformers_import()
+
+        self.model = model
+        self.tokenizer = tokenizer
+        self.config = config
+        self.engine_type = engine_type
+        self.max_length = max_length
+        self.input_names = input_names
+        self.binary_output = binary_output
+        self._args_parser = args_parser or DefaultArgumentHandler()
+        self._framework = (
+            "np" if self.engine_type in [DEEPSPARSE_ENGINE, ORT_ENGINE] else "pt"
+        )
+
+    def transform(self, X):
+        """
+        Scikit / Keras interface to transformers' pipelines.
+        This method will forward to __call__().
+        """
+        return self(X=X)
+
+    def predict(self, X):
+        """
+        Scikit / Keras interface to transformers' pipelines.
+        This method will forward to __call__().
+        """
+        return self(X=X)
+
+    def _parse_and_tokenize(
+        self, *args, padding=True, add_special_tokens=True, **kwargs
```

Comment:
Is there an engine reason for setting padding to true? IIRC if its set to false the batches will be dynamically padded to max size for that batch.

Reply:
this comes from huggingface's source code for the base pipeline. the QA pipeline doesn't follow this path, and goes through `squad_convert_examples_to_features` instead